static/docs/command-reference/run.md

```diff
@@ -54,23 +54,20 @@ try creating a pipeline.
 
 ### Well-behaved commands
 
-DVC is simple to use, you only need to wrap your commands with `dvc run`, and
-define your dependencies and outputs.
+In a sense, `dvc run` works as a special wrapper for your commands. To prevent
+unexpected behavior when DVC executes them, your commands should ideally follow
+these rules:
```

Comment:
What "unexpected behavior" are we talking about? At `dvc run` itself, or later at `dvc repro`? Will they crash or something?

Reply:
I think both - run and repro.

In case of DVC the major "unexpected behavior" that comes to my mind is inconsistency between DVC-file (checksums) and actual data. Command writes to an output of another command. Or command writes to an input to itself.

Less obvious - when files (dependencies, outputs, DVC-files) look consistent but when you run repro you get a completely different result. A missing dependency case. To some extent it can be also considered an "unexpected behavior".

Reply:
> - Inconsistency between DVC-file (checksums) and actual data: Command writes to an output of another command, or command writes to an input to itself.

Ah, I see. Al right. So basically stage/pipeline behaviors that DVC doesn't support? Maybe we should rename the section something like that because "well behaved" sounds like we have a strong opinion about how data pipelines should work in general.

> - Files look consistent  but when you run repro you get a completely different result. A missing dependency case.

I don't understand this one. Is the files including dependencies are consistent how can the dependency be missing? And if a dependency file is missing you can just `dvc pull` to restore it, no? Maybe I'm not getting the point. But anyway, my main suggestion here is the above (not sounding opinionated).

Reply:
> In case of DVC the major "unexpected behavior" that comes to my mind is inconsistency between DVC-file (checksums) and actual data.

Yes, this is the reason we are including this section.

> I don't understand this one. Is the files including dependencies are consistent how can the dependency be missing? And if a dependency file is missing you can just dvc pull to restore it, no? Maybe I'm not getting the point. But anyway, my main suggestion here is the previous one (not sounding opinionated).

@jorgeorpinel , I think @shcheklein was talking about forgetting to specify a dependency in a run command:
```
dvc run -d data -o myscript.py data
```

(you are not tracking your code, thus, `status` will not know whether DVC needs to reproduce it again or not)

Reply:
OK. I changed the title to "Unsupported command behavior" and made other small improvements, please review.